build.gradle

```diff
@@ -1,93 +1,286 @@
-ext.dockerBuildArgs = getDockerBuildArgs()
-apply from: 'gradle/docker.gradle'
+// plugins {
+//     id 'com.github.blindpirate.gogradle' version '0.8.0'
+// }
 
-
-task removeBinary(type: Delete) {
-    delete "${projectDir}/bin/wsk"
-    delete "${projectDir}/bin/mac"
-    delete "${projectDir}/bin/linux"
-    delete "${projectDir}/bin/windows"
+buildscript {
+    repositories {
+        mavenLocal()
+        jcenter()
+    }
+    dependencies {
+        classpath "org.antlr:antlr4:4.6"
+        classpath group: 'com.google.inject', name: 'guice', version: '4.1.0'
+        classpath group: 'org.apache.commons', name: 'commons-lang3', version: '3.5'
+        classpath group: 'org.apache.commons', name: 'commons-collections4', version: '4.1'
+        classpath group: 'commons-codec', name: 'commons-codec', version: '1.10'
+        classpath group: 'commons-io', name: 'commons-io', version: '2.5'
+        classpath 'com.google.guava:guava:20.0'
+        classpath 'com.github.zafarkhaja:java-semver:0.9.0'
+        classpath group: 'org.jsoup', name: 'jsoup', version: '1.7.2'
+        classpath group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: '2.8.5'
+        classpath group: 'com.fasterxml.jackson.core', name: 'jackson-annotations', version: '2.8.5'
+        classpath group: 'com.google.code.findbugs', name: 'annotations', version: '3.0.1'
+        classpath group: 'com.fasterxml.jackson.dataformat', name: 'jackson-dataformat-yaml', version: '2.8.5'
+        classpath group: 'net.lingala.zip4j', name: 'zip4j', version: '1.3.2'
+        classpath group: 'org.apache.ant', name: 'ant', version: '1.10.0'
+        classpath group: 'com.typesafe.sbt', name: 'sbt-interface', version: '0.13.13'
+        classpath group: 'com.fasterxml.jackson.dataformat', name: 'jackson-dataformat-xml', version: '2.6.3'
+        classpath group: 'org.codehaus.woodstox', name: 'woodstox-core-asl', version: '4.4.1'
+        classpath group: 'org.jooq', name: 'joor', version: '0.9.6'
+        classpath 'com.moandjiezana.toml:toml4j:0.7.2'
+        classpath files('../gogradle/build/libs/gogradle-0.8.1.jar')
+    }
 }
 
-task distBinary(dependsOn: [removeBinary, buildBinaries]) {
-    doLast {
-        run(dockerBinary + ["rm", "-f", dockerContainerName], true)
-        run(dockerBinary + ["run", "--name", dockerContainerName, dockerTaggedImageName])
+apply plugin: 'com.github.blindpirate.gogradle'
 
-        // Copy all Go binaries from Docker into openwhisk/bin folder
-        run(dockerBinary + ["cp", dockerContainerName +
-                ":/src/github.com/apache/incubator-openwhisk-cli/build/.", "${projectDir}/bin"])
+golang {
+    packagePath = 'github.com/apache/incubator-openwhisk-cli' as String
+    buildTags = (rootProject.findProperty('goTags')?:'').split(',')
+}
 
-        run(dockerBinary + ["rm", "-f", dockerContainerName])
+dependencies {
+    golang {
+        // TODO - validate where version should and should not be locked.
```

Comment:
Versions need be locked for every imported package for legal reasons and to prevent random breakages. How does this affect Godeps? Do we need to lock the commit hashes in two places now? Will running `go get` not pick these dependancies up?

Reply:
The gogradle script supports locking.  I'll put it into place for the next build cycle.  You can see the commits already in place for most imports, grabbed from godeps.  Gogradle takes over management of the locks from gogradle.  I'll take a look at #201 see that everything's buttoned up.  Will also document locking in the script.

Reply:
We need to lock the dependancies using `godep` though.